__tests__/pet.test.js

```diff
@@ -0,0 +1,179 @@
+const Pet = require('../src/pet');
+
+describe('constructor', () => {
+    it('returns an object', () => {
+      expect(new Pet('Fido')).toBeInstanceOf(Object);
+    });
+    it ('sets the name property', () => {
+      const pet = new Pet ('Prince')
+      expect(pet.name).toEqual('Prince');
+    });
+    it ('has an initial age of 0', () => {
+      const pet = new Pet ('Prince')
+      expect(pet.age).toEqual(0);
+    });
+    it ('has an initial hunger of 0', () => {
+      const pet = new Pet ('Prince')
+      expect(pet.hunger).toEqual(0);
+    });
+    it ('has an initial fitness of 10', () => {
+      const pet = new Pet ('Prince')
+      expect(pet.fitness).toEqual(10);
+
+    });
+  });
+
+  describe('isAlive', () => {
+    it('tells you if your pets fitness is 0 so has died', () => {
+
+      const pet = new Pet ('Satan')
+      pet.fitness= 0;
+
+      expect(pet.isAlive).toBe(false);
+    });
+    it('tells you if your pets hunger is 10 or more so has died', () => {
+
+      const pet = new Pet ('Satan')
+      pet.hunger= 10;
+
+      expect(pet.isAlive).toBe(false);
+    });
+    it('tells you if your pets age is 30 so has died', () => {
+
+      const pet = new Pet ('Satan')
+      pet.age= 30;
+
+      expect(pet.isAlive).toBe(false);
+    });
+
+  });
+
+  describe('growUp', () => {
+    it ('increases age by 1', () => {
+      const pet = new Pet ('Giermo')
+      pet.growUp();
+
+      expect(pet.age).toEqual(1);
+    });
+    it ('increases hunger by 5', () => {
+      const pet = new Pet ('Laslo')
+      pet.growUp();
+
+      expect(pet.hunger).toEqual(5);
+
+    });
+    it ('decreases fitness by 3', () => {
+      const pet = new Pet ('Nadya')
+      pet.growUp();
+
+      expect(pet.fitness).toEqual(7);
+    });
+    it('throws an error if the pet is not alive', () => {
+      const pet = new Pet ('Satan');
+
+      pet.age = 30;
+
+      expect(() => pet.growUp()).toThrow('Your pet is no longer alive');
+    });
+
+  });
+
+  describe ('walk', () => {
+    it ('increases fitness by 4, to a maximum of 10', () => {
```

Comment:
This name is wrong, as you are not asserting that the fitness will increment by 4, only that it wont go over a specified amount. You could fix this by setting the fitness to 2, running the walk function then asserting it is up to 6, then setting the fitness to 9, calling the walk function and asserting it didnt go over 10

Incidentally, having the values hard coded into the test names like this is not the ideal way to go about this. What if you wanted to change the fitness threshold or the amount the walk function alters fitness by in the future? You would have to rewrite all the tests and update everything twice.
A way around this would be to define the constants you did in the pet.js file in their own file, then import them into pet.js and this test file, then updating the values in that one place would change everything throughout the code and make for much more easily maintainable work

Last thing would be to change the name of the test to something more abstract like "Assert fitness level increments properly and stays within correct range"